File: 04_implementation/frontend/src/api/crew.ts
import { request } from "./client";

export type Crew = {
  row_number: number;
  codigo: string;
  descripcion: string;
  valor_dia: string;
  disponibilidad: string | null;
  valor_hora_formula: string | null;
};

export type CrewSearchResponse = {
  items: Crew[];
  total: number;
  search: string;
  can_create: boolean;
};

export function searchCrew(search: string): Promise<CrewSearchResponse> {
  const query = new URLSearchParams({ search, limit: "8" });
  return request<CrewSearchResponse>(`/crew?${query.toString()}`);
}

File: 04_implementation/backend/app/schemas/crew.py
from __future__ import annotations

from decimal import Decimal

from pydantic import BaseModel


class CrewRead(BaseModel):
    row_number: int
    codigo: str
    descripcion: str
    valor_dia: Decimal
    disponibilidad: Decimal | None = None
    valor_hora_formula: str | None = None
